karton/core/test.py

```diff
@@ -215,87 +217,16 @@ def run_task(self, task: Task) -> List[Task]:
         :param task: Task to be spawned
         :return: Result tasks sent by Karton Service
         """
-        for key, resource in task.iterate_resources():
-            if not isinstance(resource, TestResource):
-                raise TypeError(
-                    "Input resources must be instantiated using TestResource "
-                    "instead of plain LocalResource (payload key: '{}')".format(key)
-                )
-        return self.karton.run_task(task)
-
-
-class KartonMock(object):
-    identity = ""
-    filters: List[dict] = []
-
-    @classmethod
-    def from_karton(cls, karton_class):
-        """
-        Turns Consumer into its mocked version, so we can test subsystem
-        without interaction with infrastructure
-
-        :param karton_class: Consumer-based Karton class
-        :type karton_class: Type[Consumer]
-        :return: Type[KartonMock]
-        """
-        if not issubclass(karton_class, Consumer):
-            raise TypeError("Karton must be karton.Consumer subclass")
-
-        if "internal_process" in karton_class.__dict__:
-            raise TypeError("Karton can't override internal_process method")
-
-        return type(karton_class.__name__ + "Mock", (cls,), dict(karton_class.__dict__))
-
-    def __init__(self, config, **kwargs) -> None:
-        self.config = config
-        self.current_task: Optional[Task] = None
-        self._result_tasks: List[Task] = []
-
-    @property
-    def log(self) -> logging.Logger:
-        return logging.getLogger(self.identity)
-
-    def send_task(self, task: Task) -> None:
-        """Mock the normal send_task and instead just append it to an internal
-        list of output tasks
-
-        :param task: The Task object to send
-        """
-        self.log.debug("Dispatched task %s", task.uid)
-
-        # Complete information about task
-        if self.current_task is not None:
-            task.set_task_parent(self.current_task)
-            task.merge_persistent_payload(self.current_task)
-            task.priority = self.current_task.priority
+        self.karton.backend.produced_tasks = []
+        self.karton.current_task = task
 
-        task.last_update = time.time()
-        task.headers.update({"origin": self.identity})
+        if get_function_arg_num(self.karton.process) == 0:
```

Comment:
is this backward compatiblity for other karton versions?

(Maybe add a small comment, like # backward compatibility for karton <3.0)

Reply:
Both parameter passing methods are accepted right now, added a comment anyway